tests/test_decorators.py

```diff
@@ -518,3 +518,33 @@ def validate_many(self, data, many):
         errors = schema.validate([{'foo': 3, 'bar': 'not an int'}], many=True)
         assert 'bar' in errors[0]
         assert '_schema' not in errors
+
+    def test_allow_reporting_field_errors_in_schema_validator(self):
+
+        class NestedSchema(Schema):
+            baz = fields.Int(required=True)
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Nested(NestedSchema, required=True)
+            bam = fields.Int(required=True)
+
+            @validates_schema(skip_on_field_errors=True)
+            def consistency_validation(self, data):
+                errors = {}
+                if data['bar']['baz'] != data['foo']:
+                    errors['bar'] = {'baz': 'Non-matching value'}
+
+                if data['bam'] > data['foo']:
+                    errors['bam'] = 'Value should be less than foo'
+
+                if errors:
+                    raise ValidationError(errors)
+
+        schema = MySchema()
+        errors = schema.validate({'foo': 2, 'bar': {'baz': 5}, 'bam': 6})
```

Comment:
I think you are confusing the documentation for the [`message` parameter of `ValidationError`](https://marshmallow.readthedocs.io/en/dev/api_reference.html#marshmallow.exceptions.ValidationError) with the documentation for the return value of [Schema.load()](https://marshmallow.readthedocs.io/en/dev/api_reference.html#marshmallow.Schema.load). The `errors` value returned by `load()` is a named tuple, and its value is documented in the sections on [Schema Level Validation](https://marshmallow.readthedocs.io/en/dev/extending.html#schema-level-validation) and [Storing Errors on Specific Fields](https://marshmallow.readthedocs.io/en/dev/extending.html#storing-errors-on-specific-fields).

> It looks inconsistent. This PR fixes that.

The "inconsistency" between the two interfaces is fully documented. This PR is a breaking change.